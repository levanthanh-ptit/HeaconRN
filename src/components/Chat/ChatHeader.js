import React, { Component } from 'react'
import { Text, StyleSheet, View, Image } from 'react-native'

export default class ChatHeader extends Component {
    
  render() {
    return (
      <View style={Styles.HeaderContainer}>
        <Image  style={Styles.HeaderAvatar}/>
        <Text>{}</Text>
      </View>
    )
  }
}

const Styles = StyleSheet.create({
    HeaderContainer:{
        display: 'flex',
        flexDirection: 'row',
        width: '100%',
        height: '100%',
        alignItems: 'center',
    },
    HeaderAvatar:{
        height: 50,
        width: 50,
    },

})
