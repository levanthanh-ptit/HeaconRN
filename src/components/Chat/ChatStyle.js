import { StyleSheet, Dimensions } from 'react-native'
export const Styles = StyleSheet.create({
  ChatContainer: {
    width: "100%",
    display: 'flex',
    height: '100%'
  },
  ListBubbleWrapedContainer: {
    width: "100%",
    display: 'flex',
    height: '100%',
  }
}
)

