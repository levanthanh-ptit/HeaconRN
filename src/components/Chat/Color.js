export const blue = '#3578e5'
export const darkBlue = '#245099'
export const red = '#ff1010'
export const grey = '#eeeeee'
export const darkGrey = '#999999'