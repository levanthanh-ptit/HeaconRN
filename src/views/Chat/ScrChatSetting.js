import React, { Component } from 'react'
import { View } from 'react-native'
import ChatSetting from '../../components/Chat/ChatSetting/ChatSetting'
export default class ScrChatSetting extends Component {
  static navigationOptions = {
    title: 'Setting',
  };
  render() {
    return (
      <View>
        <ChatSetting></ChatSetting>
      </View>
    )
  }
}