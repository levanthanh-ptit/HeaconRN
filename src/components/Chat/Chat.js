import React, { Component } from 'react'
import { Text, View, Button, FlatList } from 'react-native'
import { Styles, BubbleColor } from './ChatStyle'
import Bubble from './Bubble/Bubble'
import TypingBar from './TypingBar/TypingBar'
import I from '../UI/AppIcon'

export default class Chat extends Component {
  state = {
    messages: null,
    selectedBubble: null
  }
  constructor(props) {
    super(props)
  }
  componentDidMount() {
    //fake date for loading from route path
  
    var mess = [];
    for (let i = 0; i < 11; i++) {
      mess.push({
        id: i + '',
        content: <Text>{'abc ' + i + (i % 3 == 0 ? 'divide perfect for 3 divide perfect for 3' : '')}</Text>,
        type: (i % 2) == 0 ? 'my' : 'other'
      });
    }
    mess.push({
      id: '11',
      content: <View><Text style={{ backgroundColor: 'red' }}>test view inside</Text></View>,
      type: 'my',
    })
    this.setState({
      messages: mess,
      selectedBubble: null,
    })
  }

  _deleteItem = (id) => {
    var mess = this.state.messages.filter(e => {
      if (e.id != id) return true
    })
    this.setState({
      messages: mess
    })
  }
  _onPressItem = (id) => {
    var id = id == this.state.selectedBubble ? null : id;
    this.setState({
      selectedBubble: id,
    })
  }

  _keyExtractor = (item, index) => item.id;

  _renderBubble = ({ item }) => {
    var selected = this.state.selectedBubble === item.id;
    return <Bubble
      key={item.id}
      id={item.id}
      type={item.type}
      onPressItem={this._onPressItem}
      selected={selected}
      onPressAction={this._deleteItem}
      ActionElement={null}
      content={item.content}
    ></Bubble>
  }

  render() {
    const buttonIconSize = 20;
    const buttonIconColor = '#666';
    var addActtionButton = {
      id: 1,
      action: null,
      content: <I name={'smile-o'} color={buttonIconColor} size={buttonIconSize}/>,
      position: 'left'
    }
    var addImageActtionButton = {
      id: 2,
      action: null,
      content: <I name={'image'} color={buttonIconColor} size={buttonIconSize}/>,
      position: 'left'
    }
    var sendActtionButton = {
      id: 3,
      action: null,
      content: <I name={'send'} color={'#DC143C'} size={buttonIconSize}/>,
      position: 'right'
    }
    return (
      <View
        id='ChatContainer'
        style={Styles.ChatContainer}
      >
        <FlatList
          id='ChatList'
          style={Styles.ListBubbleWrapedContainer}
          inverted={true}
          data={this.state.messages}
          extraData={this.state}
          renderItem={this._renderBubble}
          keyExtractor={this._keyExtractor}
        />
        <TypingBar listAction={[addImageActtionButton, addActtionButton, sendActtionButton]}></TypingBar>
      </View>
    )
  }
}

