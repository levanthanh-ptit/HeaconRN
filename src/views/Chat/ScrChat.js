import React, { Component } from 'react'
import { View } from 'react-native'
import Chat from '../../components/Chat/Chat'

export default class SrcChat extends Component {
  static navigationOptions = ({navigation}) => {
    let component = navigation.getParam('NavigationHeaderComponent', null);
    return {component}
  };
  render() {
    const { navigation } = this.props;
    var userId = navigation.getParam('userId', null);
    var targerId = navigation.getParam('tagetId', null);
    return (
      <View id='ChatContainer'>
        <Chat
          navigation={this.props.navigation}
          userId={userId}
          targerId={targerId}
        ></Chat>
      </View>
    )
  }
}
