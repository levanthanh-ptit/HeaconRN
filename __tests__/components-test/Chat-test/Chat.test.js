import React from 'react';
import Chat from '../../../src/components/Chat/Chat';

import renderer from 'react-test-renderer';

test('renders correctly', () =>{
    const tree = renderer.create(<Chat></Chat>).toJSON();
    expect(tree).toMatchSnapshot();
});