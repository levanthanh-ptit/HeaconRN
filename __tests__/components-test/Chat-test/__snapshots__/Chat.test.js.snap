// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  id="ChatContainer"
>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": "0",
          "text": "abc0",
          "type": "my",
        },
        Object {
          "id": "1",
          "text": "abc1",
          "type": "other",
        },
        Object {
          "id": "2",
          "text": "abc2",
          "type": "my",
        },
        Object {
          "id": "3",
          "text": "abc3",
          "type": "other",
        },
        Object {
          "id": "4",
          "text": "abc4",
          "type": "my",
        },
        Object {
          "id": "5",
          "text": "abc5",
          "type": "other",
        },
        Object {
          "id": "6",
          "text": "abc6",
          "type": "my",
        },
        Object {
          "id": "7",
          "text": "abc7",
          "type": "other",
        },
        Object {
          "id": "8",
          "text": "abc8",
          "type": "my",
        },
        Object {
          "id": "9",
          "text": "abc9",
          "type": "other",
        },
      ]
    }
    disableVirtualization={false}
    extraData={
      Object {
        "messages": Array [
          Object {
            "id": "0",
            "text": "abc0",
            "type": "my",
          },
          Object {
            "id": "1",
            "text": "abc1",
            "type": "other",
          },
          Object {
            "id": "2",
            "text": "abc2",
            "type": "my",
          },
          Object {
            "id": "3",
            "text": "abc3",
            "type": "other",
          },
          Object {
            "id": "4",
            "text": "abc4",
            "type": "my",
          },
          Object {
            "id": "5",
            "text": "abc5",
            "type": "other",
          },
          Object {
            "id": "6",
            "text": "abc6",
            "type": "my",
          },
          Object {
            "id": "7",
            "text": "abc7",
            "type": "other",
          },
          Object {
            "id": "8",
            "text": "abc8",
            "type": "my",
          },
          Object {
            "id": "9",
            "text": "abc9",
            "type": "other",
          },
        ],
        "selectedBubble": null,
      }
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    id="ChatList"
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "white",
        "width": "100%",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          className="GroupContainer"
          style={
            Array [
              Object {
                "alignItems": "baseline",
                "display": "flex",
                "flexDirection": "row-reverse",
                "paddingHorizontal": 10,
                "paddingVertical": 6,
                "width": "100%",
              },
            ]
          }
        >
          <View
            className="BubbleContent"
            style={
              Object {
                "display": "flex",
                "marginHorizontal": 5,
                "maxWidth": "65%",
              }
            }
          >
            <Text
              className="BubbleText"
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "#3578e5",
                    "borderRadius": 20,
                    "color": "white",
                    "padding": 15,
                  },
                ]
              }
            >
               sample text for message sample text for message sample text for message
            </Text>
          </View>
          <Text
            className="AcctionButton"
            onPress={[Function]}
            style={
              Object {
                "backgroundColor": "#eeeeee",
                "borderRadius": 25,
                "height": 50,
                "textAlign": "center",
                "textAlignVertical": "center",
                "width": 50,
              }
            }
          >
            <Text>
              Xóa
            </Text>
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          className="GroupContainer"
          style={
            Array [
              Object {
                "alignItems": "baseline",
                "display": "flex",
                "flexDirection": "row-reverse",
                "paddingHorizontal": 10,
                "paddingVertical": 6,
                "width": "100%",
              },
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            className="BubbleContent"
            style={
              Object {
                "display": "flex",
                "marginHorizontal": 5,
                "maxWidth": "65%",
              }
            }
          >
            <Text
              className="BubbleText"
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "#3578e5",
                    "borderRadius": 20,
                    "color": "white",
                    "padding": 15,
                  },
                  Object {
                    "backgroundColor": "#eeeeee",
                    "color": "black",
                  },
                ]
              }
            >
               sample text for message sample text for message sample text for message
            </Text>
          </View>
          <Text
            className="AcctionButton"
            onPress={[Function]}
            style={
              Object {
                "backgroundColor": "#eeeeee",
                "borderRadius": 25,
                "height": 50,
                "textAlign": "center",
                "textAlignVertical": "center",
                "width": 50,
              }
            }
          >
            <Text>
              Xóa
            </Text>
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          className="GroupContainer"
          style={
            Array [
              Object {
                "alignItems": "baseline",
                "display": "flex",
                "flexDirection": "row-reverse",
                "paddingHorizontal": 10,
                "paddingVertical": 6,
                "width": "100%",
              },
            ]
          }
        >
          <View
            className="BubbleContent"
            style={
              Object {
                "display": "flex",
                "marginHorizontal": 5,
                "maxWidth": "65%",
              }
            }
          >
            <Text
              className="BubbleText"
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "#3578e5",
                    "borderRadius": 20,
                    "color": "white",
                    "padding": 15,
                  },
                ]
              }
            >
               sample text for message sample text for message sample text for message
            </Text>
          </View>
          <Text
            className="AcctionButton"
            onPress={[Function]}
            style={
              Object {
                "backgroundColor": "#eeeeee",
                "borderRadius": 25,
                "height": 50,
                "textAlign": "center",
                "textAlignVertical": "center",
                "width": 50,
              }
            }
          >
            <Text>
              Xóa
            </Text>
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          className="GroupContainer"
          style={
            Array [
              Object {
                "alignItems": "baseline",
                "display": "flex",
                "flexDirection": "row-reverse",
                "paddingHorizontal": 10,
                "paddingVertical": 6,
                "width": "100%",
              },
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            className="BubbleContent"
            style={
              Object {
                "display": "flex",
                "marginHorizontal": 5,
                "maxWidth": "65%",
              }
            }
          >
            <Text
              className="BubbleText"
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "#3578e5",
                    "borderRadius": 20,
                    "color": "white",
                    "padding": 15,
                  },
                  Object {
                    "backgroundColor": "#eeeeee",
                    "color": "black",
                  },
                ]
              }
            >
               sample text for message sample text for message sample text for message
            </Text>
          </View>
          <Text
            className="AcctionButton"
            onPress={[Function]}
            style={
              Object {
                "backgroundColor": "#eeeeee",
                "borderRadius": 25,
                "height": 50,
                "textAlign": "center",
                "textAlignVertical": "center",
                "width": 50,
              }
            }
          >
            <Text>
              Xóa
            </Text>
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          className="GroupContainer"
          style={
            Array [
              Object {
                "alignItems": "baseline",
                "display": "flex",
                "flexDirection": "row-reverse",
                "paddingHorizontal": 10,
                "paddingVertical": 6,
                "width": "100%",
              },
            ]
          }
        >
          <View
            className="BubbleContent"
            style={
              Object {
                "display": "flex",
                "marginHorizontal": 5,
                "maxWidth": "65%",
              }
            }
          >
            <Text
              className="BubbleText"
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "#3578e5",
                    "borderRadius": 20,
                    "color": "white",
                    "padding": 15,
                  },
                ]
              }
            >
               sample text for message sample text for message sample text for message
            </Text>
          </View>
          <Text
            className="AcctionButton"
            onPress={[Function]}
            style={
              Object {
                "backgroundColor": "#eeeeee",
                "borderRadius": 25,
                "height": 50,
                "textAlign": "center",
                "textAlignVertical": "center",
                "width": 50,
              }
            }
          >
            <Text>
              Xóa
            </Text>
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          className="GroupContainer"
          style={
            Array [
              Object {
                "alignItems": "baseline",
                "display": "flex",
                "flexDirection": "row-reverse",
                "paddingHorizontal": 10,
                "paddingVertical": 6,
                "width": "100%",
              },
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            className="BubbleContent"
            style={
              Object {
                "display": "flex",
                "marginHorizontal": 5,
                "maxWidth": "65%",
              }
            }
          >
            <Text
              className="BubbleText"
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "#3578e5",
                    "borderRadius": 20,
                    "color": "white",
                    "padding": 15,
                  },
                  Object {
                    "backgroundColor": "#eeeeee",
                    "color": "black",
                  },
                ]
              }
            >
               sample text for message sample text for message sample text for message
            </Text>
          </View>
          <Text
            className="AcctionButton"
            onPress={[Function]}
            style={
              Object {
                "backgroundColor": "#eeeeee",
                "borderRadius": 25,
                "height": 50,
                "textAlign": "center",
                "textAlignVertical": "center",
                "width": 50,
              }
            }
          >
            <Text>
              Xóa
            </Text>
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          className="GroupContainer"
          style={
            Array [
              Object {
                "alignItems": "baseline",
                "display": "flex",
                "flexDirection": "row-reverse",
                "paddingHorizontal": 10,
                "paddingVertical": 6,
                "width": "100%",
              },
            ]
          }
        >
          <View
            className="BubbleContent"
            style={
              Object {
                "display": "flex",
                "marginHorizontal": 5,
                "maxWidth": "65%",
              }
            }
          >
            <Text
              className="BubbleText"
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "#3578e5",
                    "borderRadius": 20,
                    "color": "white",
                    "padding": 15,
                  },
                ]
              }
            >
               sample text for message sample text for message sample text for message
            </Text>
          </View>
          <Text
            className="AcctionButton"
            onPress={[Function]}
            style={
              Object {
                "backgroundColor": "#eeeeee",
                "borderRadius": 25,
                "height": 50,
                "textAlign": "center",
                "textAlignVertical": "center",
                "width": 50,
              }
            }
          >
            <Text>
              Xóa
            </Text>
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          className="GroupContainer"
          style={
            Array [
              Object {
                "alignItems": "baseline",
                "display": "flex",
                "flexDirection": "row-reverse",
                "paddingHorizontal": 10,
                "paddingVertical": 6,
                "width": "100%",
              },
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            className="BubbleContent"
            style={
              Object {
                "display": "flex",
                "marginHorizontal": 5,
                "maxWidth": "65%",
              }
            }
          >
            <Text
              className="BubbleText"
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "#3578e5",
                    "borderRadius": 20,
                    "color": "white",
                    "padding": 15,
                  },
                  Object {
                    "backgroundColor": "#eeeeee",
                    "color": "black",
                  },
                ]
              }
            >
               sample text for message sample text for message sample text for message
            </Text>
          </View>
          <Text
            className="AcctionButton"
            onPress={[Function]}
            style={
              Object {
                "backgroundColor": "#eeeeee",
                "borderRadius": 25,
                "height": 50,
                "textAlign": "center",
                "textAlignVertical": "center",
                "width": 50,
              }
            }
          >
            <Text>
              Xóa
            </Text>
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          className="GroupContainer"
          style={
            Array [
              Object {
                "alignItems": "baseline",
                "display": "flex",
                "flexDirection": "row-reverse",
                "paddingHorizontal": 10,
                "paddingVertical": 6,
                "width": "100%",
              },
            ]
          }
        >
          <View
            className="BubbleContent"
            style={
              Object {
                "display": "flex",
                "marginHorizontal": 5,
                "maxWidth": "65%",
              }
            }
          >
            <Text
              className="BubbleText"
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "#3578e5",
                    "borderRadius": 20,
                    "color": "white",
                    "padding": 15,
                  },
                ]
              }
            >
               sample text for message sample text for message sample text for message
            </Text>
          </View>
          <Text
            className="AcctionButton"
            onPress={[Function]}
            style={
              Object {
                "backgroundColor": "#eeeeee",
                "borderRadius": 25,
                "height": 50,
                "textAlign": "center",
                "textAlignVertical": "center",
                "width": 50,
              }
            }
          >
            <Text>
              Xóa
            </Text>
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          className="GroupContainer"
          style={
            Array [
              Object {
                "alignItems": "baseline",
                "display": "flex",
                "flexDirection": "row-reverse",
                "paddingHorizontal": 10,
                "paddingVertical": 6,
                "width": "100%",
              },
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            className="BubbleContent"
            style={
              Object {
                "display": "flex",
                "marginHorizontal": 5,
                "maxWidth": "65%",
              }
            }
          >
            <Text
              className="BubbleText"
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "#3578e5",
                    "borderRadius": 20,
                    "color": "white",
                    "padding": 15,
                  },
                  Object {
                    "backgroundColor": "#eeeeee",
                    "color": "black",
                  },
                ]
              }
            >
               sample text for message sample text for message sample text for message
            </Text>
          </View>
          <Text
            className="AcctionButton"
            onPress={[Function]}
            style={
              Object {
                "backgroundColor": "#eeeeee",
                "borderRadius": 25,
                "height": 50,
                "textAlign": "center",
                "textAlignVertical": "center",
                "width": 50,
              }
            }
          >
            <Text>
              Xóa
            </Text>
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
